components/data/data-services/org.wso2.micro.integrator.dataservices.core/src/main/java/org/wso2/micro/integrator/dataservices/core/description/query/MongoQuery.java

```diff
@@ -373,20 +353,18 @@ public MongoQueryResult(String query, List<InternalParam> params) throws DataSer
                     }
                     this.doUpdate(collection, opQuery, mongoParams, modifier, upsert, multi);
                     break;
-
             }
         }
 
         private Iterator<Long> doCount(MongoCollection collection, String opQuery, Object[] parameters) {
             long count;
             if (opQuery != null) {
                 if (parameters.length > 0) {
-                    count = collection.count(opQuery, parameters);
-                } else {
-                    count = collection.count(opQuery);
+                    opQuery = replacePlaceholders(opQuery, parameters);
                 }
+                count = collection.countDocuments(Document.parse(opQuery.toString()));
```

Comment:
`opQuery` is already a string. Shall we remove this `toString()`.

Reply:
Fixed with [c86c955](https://github.com/wso2/product-micro-integrator/pull/3859/commits/c86c955e1787f2f73f511eb1dab8389cd594b92e)